mythril/analysis/report.py

```diff
@@ -72,24 +82,37 @@ def as_dict(self):
         return issue
 
     def add_code_info(self, contract):
-        if self.address:
+        if self.address and isinstance(contract, SolidityContract):
             codeinfo = contract.get_source_info(
                 self.address, constructor=(self.function == "constructor")
             )
             self.filename = codeinfo.filename
             self.code = codeinfo.code
             self.lineno = codeinfo.lineno
+            self.source_mapping = codeinfo.solc_mapping
+        else:
+            self.source_mapping = self.address
 
 
 class Report:
     environment = Environment(
         loader=PackageLoader("mythril.analysis"), trim_blocks=True
     )
 
-    def __init__(self, verbose=False):
+    def __init__(self, verbose=False, source=None):
         self.issues = {}
         self.verbose = verbose
-        pass
+        self.solc_version = ""
+        self.meta = {}
+
+        if source is not None:
+            self.source = source
+        else:
+            self.source = Source()
+
+        self.source_type = self.source.source_type
```

Comment:
Shouldn't these be @property functions? If source changes self.source_type should also change right?